clear up swarm sizes and manager+worker setups

BretFisher opened the pull request:
Lot's of people will have ~5-10 servers, so let's give them more detailed info.

jpetazzo commented:
Any particular reason (when people have 5-10 servers) for not having "all managers"?

I feel like it should be #ProbablyFine either way...

(Perhaps we could give both strategies? I.e., "pick 3-5 stable nodes, but if you have less than 10 nodes total, it's OK to have all of them managers too" ?)

BretFisher commented:
Well we say in a previous slide "more than 5 managers may be getting out of hand", which is valid guidance. But I keep seeing wild setups that are ill-advised so trying to be more specific in our guidance for clarity, yet give them flexibility. **Let me take another pass at this and also add a new slide about things like:**

- Security-focused? Don't run apps on Managers, and have them on a different network that is better protected from users. Managers store secrets and have encryption keys on disk by default (to decrypt raft), so it's best to have them more protected and put workers in DMZ.
- More Managers does not mean more capacity. Only one Manager does the work at a time (the Leader). The amount of resources on the managers (CPU, Memory, Network) is what affects the capacity of your swarm. Only add more managers for redundnacy (across availibility zones or datacenter racks)

What I see in the real world... that it's sort of a sliding scale of growth, where:

People with very small needs tend to do 3 managers that do all the work. They can't justify the $$ of separating out managers.

People with slightly larger needs or are security-focused will build 3 managers that don't do work, and 3-5 workers up front in dmz.

If they are able to pay for more redundancy: then 5 manager nodes with workers up front in dmz, will scale up easily, but they often start with 5+ workers.  We should guide them against making 7 node swarms that are all managers (if you need that level of redundancy, then you likely can afford to split out the work and protect the managers).

I think the points I'm trying to make is it's a balancing act of cost, security, and redundancy. If we give them templates (like my DockerCon talk about [Baby, HA, Biz, Flexy, and Swole Swarms](https://twitter.com/BretFisher/status/989183011756363778) )

jpetazzo commented:
OK! What about this:

- 1-5 nodes: all managers
- 6-10 nodes: 3 managers
- 11+ nodes: 5 managers

Then I would also add paragraphs explaining:
- as soon as you have non-manager nodes (so, 6+ nodes), don't run services on managers (except e.g. logging, metrics, stuff like that), because security
- on small clusters, managers can be very modest machines (I was thinking suggesting 4GB RAM 2 cores is fine for 10 nodes, WDYT?)
- on big clusters with lots of tasks, adding managers doesn't help because scheduling happens on the (single) leader anyway
- on very big clusters, more managers helps to smoothen the network load

Does that sound correct to you? Anything I forgot?

BretFisher commented:
Hey @jpetazzo I cleaned this up from the Summer initial commit, added info based on docker's docs at success.docker.com and consolidated a bit. Took out recommendations for 1000-node because that's crazy and we should likely not suggest anything for that size swarm :P
